### PR TITLE
Beacon to check on status endpoint

### PR DESCRIPTION
**Pull request by shaidar**
### What does this PR do?
Beacon to manage and report the status of a server status endpoint. It fires an event when specified values in pillar don't match returned response.

Our particular use case for this is to use it in conjunction with the [django-server-status](https://pypi.org/project/django-server-status/) and verify that specified services in pillar are functional. 

However, this is not restricted to that package and can be used with others.

### What issues does this PR fix or reference?
Don't believe there's an existing issue. Wrote it to resolve a problem we were facing.

### Tests written?

No

### Commits signed with GPG?

Yes

### CODE REVIEW

**Review by rares-pop — COMMENTED** [review]
There should be more pythonic ways to do the validation - you shouldn't have to do if 'foo' in dict followed by if isinstance dict['foo'] ..., but rather if isinstance(dict.get('foo', None), ....
You do the validation but is partial (i.e. you don't check for 'comp' to be valid when you do the validation).
You say nothing about html_response.
Do you want to check if the whole configuration is valid (i.e. the 'comp' is valid in all cases)

**Comment by cachedout** [comment]
This is one of those cases where I think that CodeClimate has some very valid points. I think it might be worth refactoring this a bit to avoid the deeply nested control statements.

File: salt/beacons/http_status.py [hunk 1]
```diff
@@ -0,0 +1,158 @@
...
+            valid = False
+            messages.append('[-] Sites for {} beacon requires {}'.format(__virtualname__,
+                                                                         required_site_attributes))
+        log.debug('[+] site: %s', site)
+        log.debug('[+] settings: %s', settings)
+
+        for optional_attrs in itertools.chain(settings.get(attr, []) for attr in optional_site_attributes):
+            for item in optional_attrs:
+                cmp = item.get('comp')
+                if cmp and not cmp in comparisons:
```
Comment (cachedout) [inline salt/beacons/http_status.py]: `if cmp and cmp not in comparisons` to satisfy the linter, please.

**Review by cachedout — CHANGES_REQUESTED** [review]

**Review by cachedout — APPROVED** [review]

**Comment by max-arnold** [comment]
@shaidar Interesting feature, however I found a couple of issues:

* The configuration example uses the `content-type` key, but the actual implementation contains `site_config.pop('content_type', 'json')` (note the underscore)
* The `content_type` variable seems to be unused (has no effect)
* It seems to be impossible to check if a html response contains some text, because it is always gets decoded as json
* It looks like the `if r.raise_for_status` check should be used with parentheses `()`, because it is a function, and also moved inside the `try/except` block
* No tests
* The event format is undocumented
* No release notes

**Pull request merged**